modules/gdscript/tests/gdscript_test_runner.cpp

```diff
@@ -110,21 +112,88 @@ void init_autoloads() {
 	}
 }
 
-void init_language(const String &p_base_path) {
-	// Setup project settings since it's needed by the languages to get the global scripts.
-	// This also sets up the base resource path.
-	Error err = ProjectSettings::get_singleton()->setup(p_base_path, String(), true);
-	if (err) {
-		print_line("Could not load project settings.");
-		// Keep going since some scripts still work without this.
+void init_project_dir(const String &p_base_path, const String &p_test_name, const String &p_copy_target) {
+	Error err;
+
+	String base_path_godot_file = p_base_path.path_join("project.godot");
+
+	// Setup project settings since it's needed for the import process.
+	String project_folder = TestUtils::get_temp_path(vformat("gdscript_%s", p_test_name));
+	String project_godot_file = project_folder.path_join("project.godot");
+	String project_folder_dot_godot = project_folder.path_join(".godot");
+	String project_folder_imported = project_folder_dot_godot.path_join("imported");
+	String dot_godot_dir_backup = TestUtils::get_temp_path("gdscript_dot_godot_backup");
+
+	Ref<DirAccess> da = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+	auto remove_dir = [&da, &err](const String &p_dir_to_remove) -> Error {
+		Ref<DirAccess> deletion_da = DirAccess::open(p_dir_to_remove);
+		if (deletion_da.is_null()) {
+			return FAILED;
+		}
+
+		err = deletion_da->erase_contents_recursive();
+		if (err != OK) {
+			return err;
+		}
+		return da->remove(p_dir_to_remove);
+	};
+
+	bool first_setup = false;
+	if (!is_test_setup.has(p_test_name) || !is_test_setup[p_test_name]) {
+		// First setup.
+		is_test_setup[p_test_name] = true;
+		first_setup = true;
+		remove_dir(dot_godot_dir_backup);
+		remove_dir(project_folder);
+
+		err = da->make_dir_recursive(project_folder_imported);
+		CHECK_MESSAGE(err == OK, vformat("couldn't create dir \"%s\"", project_folder_imported));
+
+		if (FileAccess::exists(base_path_godot_file)) {
+			err = da->copy(base_path_godot_file, project_godot_file);
+			CHECK_MESSAGE(err == OK, vformat("couldn't copy project.godot"));
```

Comment:
```suggestion
			CHECK_MESSAGE(err == OK, "Couldn't copy project.godot");
```
No `vformat` when not needed